### PR TITLE
Increase connection timeout and error info

### PR DESCRIPTION
**Pull request by lwrubel**
## Why was this change made? 🤔
Add info to several errors and adding a 409 error. Bumping the timeout in Faraday request options seemed to greatly reduce the number of timeouts noted in testing quick-marc updates. 

## How was this change tested? 🤨
Unit, testing with DSA QA deploys.

⚡ ⚠ If this change has cross service impact, including data writes to shared file systems, ***run [integration tests](https://github.com/sul-dlss/infrastructure-integration-test)*** and/or test in [stage|qa] environment, in addition to specs. ⚡

### CODE REVIEW

File: lib/folio_client.rb [hunk 1]
```diff
@@ -120,7 +123,8 @@ def put(path, body = nil, content_type: "application/json")
   def connection
     @connection ||= Faraday.new(
       url: config.url,
-      headers: DEFAULT_HEADERS.merge(config.okapi_headers || {})
+      headers: DEFAULT_HEADERS.merge(config.okapi_headers || {}),
+      request: {timeout: 120}
```
Comment (lwrubel) [inline lib/folio_client.rb:127]: I also tried setting this to `60`, but `120` was where it made a difference.

Comment (lwrubel) [reply lib/folio_client.rb:127]: Shelley and I looked at logs and it's possible kafka is continuing to be the bottleneck.

Comment (mjgiarlo) [inline lib/folio_client.rb:127]: I wonder if we should consider making `120` a default value that can be injected by consumers at initialization time?

File: lib/folio_client/unexpected_response.rb [hunk 2]
```diff
@@ -12,10 +12,12 @@ def self.call(response)
         raise ForbiddenError, "The operation requires privileges which the client does not have: #{response.body}"
       when 404
         raise ResourceNotFound, "Endpoint not found or resource does not exist: #{response.body}"
+      when 409
+        raise ConflictError, "Resource cannot be updated: #{response.body}"
```
Comment (mjgiarlo) [inline lib/folio_client/unexpected_response.rb:16]: This looks uncovered which is making CodeClimate cranky. If you're open to it, you could prolly toss a quick 409 test in the authenticator spec which is currently covering the various exception classes. (Not the best place, maybe, but it's what we have now.)

**Review by edsu — COMMENTED** [review]
Awesome to see the ConflictError being raised in a recognizable way!

File: lib/folio_client.rb (hunk 1 above)
Comment (lwrubel) [reply lib/folio_client.rb:127]: That all makes sense, will take a look at making it a default instead.

File: lib/folio_client/unexpected_response.rb (hunk 2 above)
Comment (mjgiarlo) [reply lib/folio_client/unexpected_response.rb:16]: I think we could refactor that at some point, for sure.

Comment (lwrubel) [reply lib/folio_client/unexpected_response.rb:16]: Added a spec.

File: lib/folio_client.rb [hunk 3]
```diff
@@ -44,15 +47,15 @@ class << self
     # @param url [String] the folio API URL
     # @param login_params [Hash] the folio client login params (username:, password:)
     # @param okapi_headers [Hash] the okapi specific headers to add (X-Okapi-Tenant:, User-Agent:)
-    def configure(url:, login_params:, okapi_headers:)
-      instance.config = OpenStruct.new(url: url, login_params: login_params, okapi_headers: okapi_headers, token: nil)
+    def configure(url:, login_params:, okapi_headers:, timeout: default_timeout)
```
Comment (lwrubel) [inline lib/folio_client.rb:50]: Adding a configurable timeout, with a default_timeout value.

File: lib/folio_client.rb (hunk 1 above)
Comment (lwrubel) [reply lib/folio_client.rb]: Added a default_timeout. Thanks for the suggestion, @mjgiarlo and @edsu.

File: spec/folio_client/authenticator_spec.rb [hunk 4]
```diff
@@ -73,6 +73,15 @@
       end
     end
 
+    context "when the service returns a 409 conflict error" do
+      let(:http_status) { 409 }
+      let(:http_body) { "{\"error\" : \"get bent\"}" }
```
Comment (mjgiarlo) [inline spec/folio_client/authenticator_spec.rb:78]: Folio is so rude!

File: spec/folio_client/authenticator_spec.rb [hunk 5]
```diff
@@ -73,6 +73,15 @@
...
+
+      it "raises a service unavailable exception" do
```
Comment (mjgiarlo) [inline spec/folio_client/authenticator_spec.rb:80]: copypasta?

**Review by mjgiarlo — APPROVED** [review]

File: spec/folio_client/authenticator_spec.rb (hunk 5 above)
Comment (lwrubel) [reply spec/folio_client/authenticator_spec.rb]: 🤦‍♀️ yes, fixed!

File: spec/folio_client/authenticator_spec.rb (hunk 4 above)
Comment (lwrubel) [reply spec/folio_client/authenticator_spec.rb:78]: Why should it be more polite with this error than any other? 😆

Comment (mjgiarlo) [reply spec/folio_client/authenticator_spec.rb:78]: At least it's consistent 🕺🏻

File: spec/folio_client/authenticator_spec.rb (hunk 5 above)
Comment (mjgiarlo) [reply spec/folio_client/authenticator_spec.rb]: something something stones and glass houses

**Pull request merged**